### PR TITLE
#patch Update FromWorkflowModel API

### PR DESCRIPTION
**Pull request by squiishyy**
# TL;DR
Update FromWorkflowModel to include TypedInterface in response object

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://unionai.atlassian.net/browse/CLOUD-1177

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/repositories/transformers/workflow.go
Comment (katrogan) [inline pkg/repositories/transformers/workflow.go:52]: no need to check nil here, len should be sufficient! https://pkg.go.dev/builtin#len

File: pkg/repositories/transformers/workflow.go
Comment (katrogan) [inline pkg/repositories/transformers/workflow.go:55]:
```suggestion
			return admin.Workflow{}, errors.NewFlyteAdminErrorf(codes.Internal, fmt.Sprintf("failed to unmarshal workflow %v interface", workflowModel.ID))
```
since this includes inputs and outputs

File: pkg/repositories/transformers/workflow_test.go
Comment (katrogan) [inline pkg/repositories/transformers/workflow_test.go:76]: no need to unmarshal for this test, you can always re-use `testutils.GetWorkflowRequest().Spec.Template.Interface`
from https://github.com/flyteorg/flyteadmin/blob/dc8cc9d32b30e82c7d147a1ac88377a41165b195/pkg/manager/impl/testutils/mock_requests.go#L98-L113

File: pkg/repositories/transformers/workflow_test.go
Comment (katrogan) [inline pkg/repositories/transformers/workflow_test.go:170]: nit: I think you don't need to initialize this interface if it's just going to be nil anyways

**Review by katrogan — COMMENTED** [review]
looks great! let's make sure to test end to end before merging

File: pkg/repositories/transformers/workflow_test.go
Comment (squiishyy) [reply pkg/repositories/transformers/workflow_test.go:76]: Makes sense, will fix

File: pkg/repositories/transformers/workflow.go
Comment (katrogan) [inline pkg/repositories/transformers/workflow.go:55]: nit can we wrap the marshal err in here too?

**Review by katrogan — COMMENTED** [review]
looks good! you may need to follow the instructions for DCO sign-off

**Review by katrogan — APPROVED** [review]

**Pull request merged**
